amr-wind/equation_systems/DiffusionOps.H

```diff
@@ -149,10 +149,33 @@ struct DiffusionOp<
         auto tau_state = std::is_same<Scheme, fvm::Godunov>::value
                              ? FieldState::New
                              : fstate;
+
         amrex::MLMG mlmg(*this->m_applier);
         mlmg.apply(
             this->m_pdefields.diff_term.state(tau_state).vec_ptrs(),
             this->m_pdefields.field.vec_ptrs());
+
+        const auto& repo = this->m_pdefields.repo;
+        const int nlevels = repo.num_active_levels();
+        auto& divtau = this->m_pdefields.diff_term.state(tau_state);
+        auto& m_density = this->m_pdefields.repo.get_field("density");
+        const auto& density = m_density.state(FieldState::NPH);
+        if (!PDE::multiply_rho) {
+            for (int lev = 0; lev < nlevels; ++lev) {
+                const auto& divtau_arrs = divtau(lev).arrays();
+                const auto& rho_arrs = density(lev).const_arrays();
+
+                amrex::ParallelFor(
+                    divtau(lev), [=] AMREX_GPU_DEVICE(
```

Comment:
I think it would be worth making this a component loop as well.`divtau.nComp()` and then `divtau_arrs[nbx](i, j, k, n)`.

Reply:
Yeah, absolutely.